### PR TITLE
signv2: Do not use path encoding for query values.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Use query escape instead for comparing signature.

## Motivation and Context
Reported by community member @bsideup

## How Has This Been Tested?
Manually and with help from @vadmeste 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bsideup** [comment]
I can test this change once it's merged and deployed as `edge` tag :+1:

**Comment by bsideup** [comment]
@harshavardhana actually, just checked locally by building an image and it works! 💯

File: cmd/signature-v2.go [hunk 1]
```diff
@@ -128,12 +130,13 @@ func doesPresignV2SignatureMatch(r *http.Request) APIErrorCode {
...
 	}
 
+	// Check if the presigned URL has expired.
 	if expiresInt < time.Now().UTC().Unix() {
 		return ErrExpiredPresignRequest
 	}
 
 	expectedSignature := preSignatureV2(r.Method, encodedResource, strings.Join(filteredQueries, "&"), r.Header, expires)
-	if gotSignature != getURLEncodedName(expectedSignature) {
+	if gotSignature != url.QueryEscape(expectedSignature) {
```
Comment (vadmeste) [inline cmd/signature-v2.go]: If the client is free to encode or not to encode query values, this change will be wrong: The client can send here a raw '/' without escaping it and this comparaison line won't work. So, if we want to support both escaped and raw query values, we can use url.QueryUnescape(gotSignature) instead.

Comment (krishnasrinivas) [reply cmd/signature-v2.go]: http://docs.aws.amazon.com/general/latest/gr/signature-version-2.html says:

`The resulting signature must be base-64 encoded and then URI encoded.`

so this looks fine

**Review by krishnasrinivas — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]

**Pull request merged**